sql/parser/eval.go

```diff
@@ -259,6 +263,156 @@ func (d DString) String() string {
 	return StrVal(d).String()
 }
 
+// maxTime and minTime are not the theoretical max/min for the time.Time
+// type, but are sufficiently into the future/past.
+const (
+	hundredBillion = 100 * 1000 * 1000 * 1000
+)
+
+var maxTime = time.Date(hundredBillion, time.January, 1, 0, 0, 0, 0, time.UTC)
```

Comment:
The year argument to `time.Date` is an `int`, so this won't compile on any system where `int != int64`. Perhaps change 100b to 1b. I'm struggling to imagine why the extra range is necessary.